### PR TITLE
multiple changes from my fork

### PR DESCRIPTION
**Pull request by fiatjaf**
Sorry for bundling these all, we can unbundle them later if you think only some parts are good enough.

This PR contain the following changes:

1. Recognize and insert links using the new `[[[...]]]` syntax instead of `<...>`;
2. Replace `:` with `-` in zettel titles search view so vim/fzf don't break these;
3. `gzi`/`gzl` insert at the cursor, not in their own line followed by a newline;
4. `gzl` doesn't start a new zettel in insert mode, start in normal mode;
5. `gzu/gzl` go back to the previously visited zettel instead of the previously edited zettel.

### CODE REVIEW

File: ftdetect/neuron.vim [hunk 1]
```diff
@@ -1,6 +1,17 @@
 let g:zextension = get(g:, 'zextension', '.md')
 let g:zkdir = get(g:, 'zkdir', $HOME.'/zettelkasten/')
 
+" search for neuron.dhall
+let s:current = expand("%:p")
+let s:dir = fnamemodify(s:current, ":h:r")
+while s:dir != "/"
```
Comment (agundy) [inline ftdetect/neuron.vim:7]: This leaves the editor hanging searching for a very long time on my setup. Is there a more efficient way to do what you need here? I'm not actual sure what the neuron.dhall file is.

**Comment by fiatjaf** [comment]
Please evaluate each commit separately or this will look like a big mess.

File: ftdetect/neuron.vim (hunk 1 above)
Comment (fiatjaf) [reply ftdetect/neuron.vim:7]: Very long? It should be done in a few milliseconds. What happened when you tried to run?

That file holds settings for the site.

Comment (agundy) [reply ftdetect/neuron.vim:7]: Strange, I just tried again and it's fast now. The symptom I saw was booting Vim in my non-zettelkasten dir and being shown a blank screen for a long time. When I ctrl-c'ed the line reported being interrupted was 13 of this file.

Comment (srid) [reply ftdetect/neuron.vim:7]: The `neuron query` command can be slow in large Zettelkastens. See https://github.com/srid/neuron/issues/321

Comment (fiatjaf) [reply ftdetect/neuron.vim:7]: If you're using my latest `master`. That is caused by the initial `refresh_cache` call. I've done it with an async job in neovim, but in vim the async job is causing a weird UI bug I'm yet to fix, so I temporarily make it run in sync mode.

Comment (fiatjaf) [reply ftdetect/neuron.vim:7]: Fixed in latest master, but even with async jobs there's a delay. Thinking about how to prevent that.

Comment (srid) [reply ftdetect/neuron.vim:7]: Pretty sure that's because of https://github.com/srid/neuron/issues/321

Every instance of `neuron query` builds the entire graph from disk; it needs to be optimized, so that only modified files get loaded, while graph cache (cache.json) is used for the rest.

File: README.md [hunk 2]
```diff
@@ -2,47 +2,57 @@
...
 - [ripgrep](https://github.com/BurntSushi/ripgrep)
-- [jq](https://stedolan.github.io/jq/)
 
 
 ## Installation
 ### Using [vim-plug](https://github.com/junegunn/vim-plug)
 ```vim
 Plug 'junegunn/fzf.vim'
 Plug 'BurntSushi/ripgrep'
+Plug 'fiatjaf/neuron.vim'
```
Comment (MaienM) [inline README.md:18]: This shouldn't be part of this PR, right?

File: README.md [hunk 3]
```diff
@@ -2,47 +2,57 @@
...
 nm gzr <Plug>NeuronRefreshCache
-nm gzo <Plug>EditZettelUnderCursor
 ```
 You can disable the mappings with letting the `g:neuron_no_mappings` variable to
 1:
 ```vim
 let g:neuron_no_mappings = 1
 ```
 
+## Differences from `ihsanturk/neuron.vim`
```
Comment (MaienM) [inline README.md:42]: Same here, this block makes no sense if this were to be merged.

File: autoload/util.vim [hunk 4]
```diff
@@ -1,136 +1,73 @@
...
-	if l:n < 0 " index given as optional arg in
-		return l:found " list
-	else
-		return l:found[l:n] " string
-	end
+	return ""
 endf
 
 func! util#deform_zettelid(zettelid)
-	if a:zettelid =~ "<.*>"
```
Comment (MaienM) [inline autoload/util.vim:79]: Should this branch not also be kept? The old "autolink" syntax is still valid, and some people will probably still have it in their notes.

File: README.md (hunk 2 above)
Comment (fiatjaf) [reply README.md:18]: Right.

File: autoload/util.vim (hunk 4 above)
Comment (fiatjaf) [reply autoload/util.vim:79]: I didn't think that would be the case.

File: ftdetect/neuron.vim (hunk 1 above)
Comment (fenetikm) [reply ftdetect/neuron.vim:7]: my editor hangs too - if that file isn't found it goes forever it seems.

**Comment by fiatjaf** [comment]
I guess this is not mergeable anymore.

File: ftdetect/neuron.vim (hunk 1 above)
Comment (fiatjaf) [reply ftdetect/neuron.vim:7]: I can't reproduce that. Tried with `vim` and `nvim`.
@fenetikm, does it work if you quit and restart vim like it did for @agundy above?
Can you create an issue on my fork?

I'll close this PR as it is not a real PR anymore.

Comment (fenetikm) [reply ftdetect/neuron.vim:7]: yep, will try to investigate and create an issue on your fork 👍

**Comment by fenetikm** [comment]
@fiatjaf you will need to enable issues on your fork 😄

**Comment by fiatjaf** [comment]
Oops, now they are enabled!
